src/dawn/engine_dawn.cc

```diff
@@ -14,15 +14,143 @@
 
 #include "src/dawn/engine_dawn.h"
 
+#include <algorithm>
 #include <utility>
 #include <vector>
 
 #include "dawn/dawncpp.h"
 #include "src/dawn/device_metal.h"
+#include "src/sleep.h"
 
 namespace amber {
 namespace dawn {
 
+namespace {
+
+// The dimensions of the framebuffer.  The code also assumes the framebuffer
+// is 2D and does not have any array layers.
+enum { kFramebufferWidth = 250, kFramebufferHeight = 250 };
+// The pipeline output color attachment slot expected for the framebuffer.
+enum { kFramebufferSlot = 0 };
+// The minimum multiple row pitch observed on Dawn on Metal.  Increase this
+// as needed for other Dawn backends.
+enum { kImageRowPitch = 256 };
+const auto kFramebufferFormat = ::dawn::TextureFormat::R8G8B8A8Unorm;
+
+// Creates a device-side texture for the framebuffer, and returns it through
+// |result_ptr|. Returns a result code.
+Result MakeFramebufferTexture(const ::dawn::Device& device,
```

Comment:
It's doing an allocation of device memory, so it can fail.  I'll add a TODO to handle that case.

It looks like Dawn needs an error callback to get that error though.  It's a bit sideways.